### ISSUE TITLE
Cannot find factory partition

### ISSUE THREAD

**Issue by pastranaluis** [gatech-sysml/modin#42]
When we try to read a csv or use any modin-gpu functionality: The following error appears

FactoryNotFoundError: Cannot find a factory for partition 'Cudf' and execution engine 'Ray'. Potential reason might be incorrect environment variable value for MODIN_BACKEND or MODIN_ENGINE

**Comment by kvu35** [comment]
So it seems that we forgot to add the appropriate factory in `modin/data_management/factories.py` so maybe adding the following lines in will help.

```python
class CudfOnRayFactory(BaseFactory):
    @classmethod
    def prepare(cls):
        """
        Fills in .io_cls class attribute lazily
        """
        from modin.engines.ray.cudf_on_ray.io import cuDFOnRayIO

        cls.io_cls = cuDFOnRayIO
```

**Comment by pastranaluis** [comment]
Yes that lines help. However, some stuff is breaking as well. I am trying to push a PR with all the changes needed to make modin-gpu work. As well as a couple other issue that will arise from it.
